search/geocoder.cpp

```diff
@@ -1095,7 +1095,7 @@ void Geocoder::GreedilyMatchStreets()
       if (IsStreetSynonymPrefix(token))
         continue;
 
-      if (feature::IsHouseNumber(token))
+      if (house_numbers::LooksLikeHouseNumber(token, true /* isPrefix */))
```

Comment:
Please, set up isPrefix correctly here. I.e. you need to check that curToken is the last token and last token is prfix.